pkg/cmd/roachtest/main_test.go

```diff
@@ -228,12 +414,59 @@ func getTestSelectionMockData() ([]registry.TestSpec, *sqlmock.Rows) {
 		{"t_randomized", "no", "5", "no"},
 	}
 	// rows are the rows that are returned by snowflake. The list of string represents the columns of each row
-	rows := sqlmock.NewRows([]string{
-		"name", "selected", "avg_duration", "last_failure_is_preempt",
-	})
+	rows := sqlmock.NewRows(testselector.AllRows)
 	// rows are populated with the data
 	for _, ds := range data {
 		rows.FromCSVString(strings.Join(ds, ","))
 	}
 	return specs, rows
 }
+
+const charset = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
+
+// generateRandomString generates a random string of a given length
+func generateRandomString(n int) (string, error) {
+	result := make([]byte, n)
+	for i := range result {
+		r, _ := randutil.NewTestRand()
```

Comment:
Hmm, actually that's not true, it just generates a new seed off of it so it will keep giving you a new seed deterministically. I still think it makes sense to pass the rng instance since every time you call this it incurs a mutex lock.